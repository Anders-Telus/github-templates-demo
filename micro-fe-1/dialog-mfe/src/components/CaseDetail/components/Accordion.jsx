import React, { useState } from 'react'
import PropTypes from 'prop-types'
import AccordionSection from './AccordionSection'
import { AccordionMenu } from '../styles'

const Accordion = ({
  accordionHeight,
  allowMultipleOpen,
  handlerAccordionOpen,
  children,
  listRef,
  isRTSPopUpOpen
}) => {
  const [openSections, setOpenSections] = useState(
    children.reduce((result, child) => {
      if (child.props.isOpen) {
        const openSectionsAcc = Object.assign({}, result)
        openSectionsAcc[child.props.label] = true
        return openSectionsAcc
      }
      return result
    }, {})
  )

  const onClick = (label) => {
    const isOpen = !!openSections[label]
    if (allowMultipleOpen) {
      setOpenSections({ ...openSections, [label]: !isOpen })
    } else {
      setOpenSections({ [label]: !isOpen })
    }
    handlerAccordionOpen({ taskId: +label, isOpen: !isOpen })
  }

  return (
    <AccordionMenu>
      {children.map((child) => {
        const {
          label, header, taskStatus, caseCompleteCancelItem
        } = child.props
        return (
          <AccordionSection
            id="accordion_section"
            isOpen={!!openSections[label]}
            onClick={() => onClick(label)}
            key={label}
            header={header}
            taskStatus={taskStatus}
            accordionHeight={accordionHeight}
            listRef={listRef}
            caseCompleteCancelItem={caseCompleteCancelItem}
            isRTSPopUpOpen={isRTSPopUpOpen}
          >
            {child.props.children}
          </AccordionSection>
        )
      })}
    </AccordionMenu>
  )
}

Accordion.defaultProps = {
  allowMultipleOpen: false,
  handlerAccordionOpen: () => {},
  accordionHeight: ''
}

Accordion.propTypes = {
  children: PropTypes.instanceOf(Object).isRequired,
  allowMultipleOpen: PropTypes.bool,
  handlerAccordionOpen: PropTypes.func,
  accordionHeight: PropTypes.string,
  listRef: PropTypes.object.isRequired,
  isRTSPopUpOpen: PropTypes.bool.isRequired
}

export default Accordion
