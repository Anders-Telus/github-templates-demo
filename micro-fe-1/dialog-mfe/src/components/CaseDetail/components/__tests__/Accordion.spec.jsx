import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from '../Accordion'

describe('Accordion test cases', () => {
  it('Should match Accordion snapshot', () => {
    const wrapper = mount(
      <Accordion>
        <div
          label="123" key="123" isOpen={true}
        >
          <div>Hello Accordion</div>
        </div>
        <div
          label="123" key="123" isOpen={true}
        >
          <div>Hello Accordion</div>
        </div>
      </Accordion>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Should Simulate the Click event ', () => {
    const wrapper = mount(
      <Accordion>
        <div
          label="123" key="123" isOpen={true}
        >
          <div>Hello Accordion</div>
        </div>
        <div
          label="1234" key="1234" isOpen={true}
        >
          <div>Hello Accordion</div>
        </div>
      </Accordion>
    )
    wrapper.find('#accordion_row_header').at(wrapper.find('#accordion_row_header').length - 1).simulate('click')
  })
  it('Should Simulate the Click event with single accordion open ', () => {
    const wrapper = mount(
      <Accordion allowMultipleOpen>
        <div
          label="123" key="123" isOpen={true}
        >
          <div>Hello Accordion</div>
        </div>
        <div
          label="1234" key="1234" isOpen={true}
        >
          <div>Hello Accordion</div>
        </div>
      </Accordion>
    )
    wrapper.find('#accordion_row_header').at(wrapper.find('#accordion_row_header').length - 1).simulate('click')
  })
  it('Should allow tooltip and allow multiple accordion open', () => {
    const wrapper = mount(
      <Accordion allowMultipleOpen>
        <div
          label="123" key="123" isOpen={true}
        >
          <div>Hello Accordion</div>
        </div>
        <div
          label="1234" key="1234" isOpen={true}
        >
          <div>Hello Accordion</div>
        </div>
      </Accordion>
    )
    wrapper.find('#accordion_row_header').at(wrapper.find('#accordion_row_header').length - 1).simulate('click')
  })
})
