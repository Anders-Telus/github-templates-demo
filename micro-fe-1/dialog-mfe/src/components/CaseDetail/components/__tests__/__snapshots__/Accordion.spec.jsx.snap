// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion test cases Should match Accordion snapshot 1`] = `
<Accordion
  accordionHeight=""
  allowMultipleOpen={false}
  handlerAccordionOpen={[Function]}
>
  <styles__AccordionMenu>
    <div
      className="styles__AccordionMenu-sc-19qwpvk-16 gVTxuZ"
    >
      <AccordionSection
        accordionHeight=""
        id="accordion_section"
        isOpen={true}
        key="123"
        onClick={[Function]}
      >
        <styles__Container>
          <div
            className="styles__Container-sc-19qwpvk-61 kJcAyc"
          >
            <FlexGrid
              gutter={false}
              limitWidth={true}
              outsideGutter={true}
            >
              <FlexGrid__StyledGrid
                limitWidth={true}
                outsideGutter={true}
                reverseLevel={
                  Array [
                    false,
                    false,
                    false,
                    false,
                    false,
                  ]
                }
              >
                <div
                  className="FlexGrid__StyledGrid-sc-1da2tb5-0 fPPQij"
                >
                  <Row>
                    <Row__StyledRow
                      reverseLevel={
                        Array [
                          false,
                          false,
                          false,
                          false,
                          false,
                        ]
                      }
                    >
                      <div
                        className="Row__StyledRow-sc-1goz0ht-0 clxkKf"
                      >
                        <Col
                          lg={1}
                          md={1}
                          sm={1}
                          xs={1}
                        >
                          <Col__StyledCol
                            gutter={false}
                            hiddenLevel={
                              Array [
                                1,
                                1,
                                1,
                                1,
                                1,
                              ]
                            }
                            horizontalAlignLevel={
                              Array [
                                "inherit",
                                "inherit",
                                "inherit",
                                "inherit",
                                "inherit",
                              ]
                            }
                            lg={1}
                            md={1}
                            sm={1}
                            xs={1}
                          >
                            <div
                              className="Col__StyledCol-sc-15yvjc7-0 gBBZLu"
                            >
                              <styles__IconWrapper>
                                <div
                                  className="styles__IconWrapper-sc-19qwpvk-62 kILCFf"
                                >
                                  <NotificationWarning>
                                    <FeedbackIcon
                                      copyDictionary={
                                        Object {
                                          "en": Object {
                                            "a11yText": "Warning",
                                          },
                                          "fr": Object {
                                            "a11yText": "Avertissement",
                                          },
                                        }
                                      }
                                      height={20}
                                      optionalText={false}
                                      viewBox="0 0 20 20"
                                      width={20}
                                    >
                                      <FeedbackIcon__StyledSVG
                                        height="1.25rem"
                                        role="img"
                                        viewBox="0 0 20 20"
                                        width="1.25rem"
                                      >
                                        <svg
                                          className="FeedbackIcon__StyledSVG-sc-1swr281-0 ldLYtC"
                                          height="1.25rem"
                                          role="img"
                                          viewBox="0 0 20 20"
                                          width="1.25rem"
                                        >
                                          <path
                                            d="M10.878 1.61l8.315 15.244a1 1 0 0 1-.878 1.48H1.685a1 1 0 0 1-.878-1.48L9.122 1.61a1 1 0 0 1 1.756 0zM10 16.794c.46 0 .833-.402.833-.898 0-.495-.373-.897-.833-.897-.46 0-.833.402-.833.897 0 .496.373.898.833.898zm-.022-2.885c.347 0 .63-.297.64-.67l.179-6.698c.01-.388-.28-.709-.64-.709h-.35c-.361 0-.65.32-.64.708l.171 6.699c.01.373.294.67.64.67z"
                                            fill="#8C5415"
                                            fillRule="evenodd"
                                          />
                                        </svg>
                                      </FeedbackIcon__StyledSVG>
                                    </FeedbackIcon>
                                  </NotificationWarning>
                                </div>
                              </styles__IconWrapper>
                            </div>
                          </Col__StyledCol>
                        </Col>
                        <Col
                          lg={11}
                          md={11}
                          sm={11}
                          xs={11}
                        >
                          <Col__StyledCol
                            gutter={false}
                            hiddenLevel={
                              Array [
                                11,
                                11,
                                11,
                                11,
                                11,
                              ]
                            }
                            horizontalAlignLevel={
                              Array [
                                "inherit",
                                "inherit",
                                "inherit",
                                "inherit",
                                "inherit",
                              ]
                            }
                            lg={11}
                            md={11}
                            sm={11}
                            xs={11}
                          >
                            <div
                              className="Col__StyledCol-sc-15yvjc7-0 diYtSO"
                            >
                              <styles__AccordionRow
                                id="accordion_row_header"
                                isOpen={true}
                                onClick={[Function]}
                              >
                                <div
                                  className="styles__AccordionRow-sc-19qwpvk-17 Wchtj"
                                  id="accordion_row_header"
                                  onClick={[Function]}
                                >
                                  <styles__AccordionColWrapper>
                                    <div
                                      className="styles__AccordionColWrapper-sc-19qwpvk-24 cVJzO"
                                    />
                                  </styles__AccordionColWrapper>
                                </div>
                              </styles__AccordionRow>
                              <styles__AccordionChildrenSection
                                accordionHeight=""
                              >
                                <div
                                  className="styles__AccordionChildrenSection-sc-19qwpvk-18 djglXG"
                                >
                                  <div>
                                    Hello Accordion
                                  </div>
                                </div>
                              </styles__AccordionChildrenSection>
                            </div>
                          </Col__StyledCol>
                        </Col>
                      </div>
                    </Row__StyledRow>
                  </Row>
                </div>
              </FlexGrid__StyledGrid>
            </FlexGrid>
          </div>
        </styles__Container>
      </AccordionSection>
      <AccordionSection
        accordionHeight=""
        id="accordion_section"
        isOpen={true}
        key="123"
        onClick={[Function]}
      >
        <styles__Container>
          <div
            className="styles__Container-sc-19qwpvk-61 kJcAyc"
          >
            <FlexGrid
              gutter={false}
              limitWidth={true}
              outsideGutter={true}
            >
              <FlexGrid__StyledGrid
                limitWidth={true}
                outsideGutter={true}
                reverseLevel={
                  Array [
                    false,
                    false,
                    false,
                    false,
                    false,
                  ]
                }
              >
                <div
                  className="FlexGrid__StyledGrid-sc-1da2tb5-0 fPPQij"
                >
                  <Row>
                    <Row__StyledRow
                      reverseLevel={
                        Array [
                          false,
                          false,
                          false,
                          false,
                          false,
                        ]
                      }
                    >
                      <div
                        className="Row__StyledRow-sc-1goz0ht-0 clxkKf"
                      >
                        <Col
                          lg={1}
                          md={1}
                          sm={1}
                          xs={1}
                        >
                          <Col__StyledCol
                            gutter={false}
                            hiddenLevel={
                              Array [
                                1,
                                1,
                                1,
                                1,
                                1,
                              ]
                            }
                            horizontalAlignLevel={
                              Array [
                                "inherit",
                                "inherit",
                                "inherit",
                                "inherit",
                                "inherit",
                              ]
                            }
                            lg={1}
                            md={1}
                            sm={1}
                            xs={1}
                          >
                            <div
                              className="Col__StyledCol-sc-15yvjc7-0 gBBZLu"
                            >
                              <styles__IconWrapper>
                                <div
                                  className="styles__IconWrapper-sc-19qwpvk-62 kILCFf"
                                >
                                  <NotificationWarning>
                                    <FeedbackIcon
                                      copyDictionary={
                                        Object {
                                          "en": Object {
                                            "a11yText": "Warning",
                                          },
                                          "fr": Object {
                                            "a11yText": "Avertissement",
                                          },
                                        }
                                      }
                                      height={20}
                                      optionalText={false}
                                      viewBox="0 0 20 20"
                                      width={20}
                                    >
                                      <FeedbackIcon__StyledSVG
                                        height="1.25rem"
                                        role="img"
                                        viewBox="0 0 20 20"
                                        width="1.25rem"
                                      >
                                        <svg
                                          className="FeedbackIcon__StyledSVG-sc-1swr281-0 ldLYtC"
                                          height="1.25rem"
                                          role="img"
                                          viewBox="0 0 20 20"
                                          width="1.25rem"
                                        >
                                          <path
                                            d="M10.878 1.61l8.315 15.244a1 1 0 0 1-.878 1.48H1.685a1 1 0 0 1-.878-1.48L9.122 1.61a1 1 0 0 1 1.756 0zM10 16.794c.46 0 .833-.402.833-.898 0-.495-.373-.897-.833-.897-.46 0-.833.402-.833.897 0 .496.373.898.833.898zm-.022-2.885c.347 0 .63-.297.64-.67l.179-6.698c.01-.388-.28-.709-.64-.709h-.35c-.361 0-.65.32-.64.708l.171 6.699c.01.373.294.67.64.67z"
                                            fill="#8C5415"
                                            fillRule="evenodd"
                                          />
                                        </svg>
                                      </FeedbackIcon__StyledSVG>
                                    </FeedbackIcon>
                                  </NotificationWarning>
                                </div>
                              </styles__IconWrapper>
                            </div>
                          </Col__StyledCol>
                        </Col>
                        <Col
                          lg={11}
                          md={11}
                          sm={11}
                          xs={11}
                        >
                          <Col__StyledCol
                            gutter={false}
                            hiddenLevel={
                              Array [
                                11,
                                11,
                                11,
                                11,
                                11,
                              ]
                            }
                            horizontalAlignLevel={
                              Array [
                                "inherit",
                                "inherit",
                                "inherit",
                                "inherit",
                                "inherit",
                              ]
                            }
                            lg={11}
                            md={11}
                            sm={11}
                            xs={11}
                          >
                            <div
                              className="Col__StyledCol-sc-15yvjc7-0 diYtSO"
                            >
                              <styles__AccordionRow
                                id="accordion_row_header"
                                isOpen={true}
                                onClick={[Function]}
                              >
                                <div
                                  className="styles__AccordionRow-sc-19qwpvk-17 Wchtj"
                                  id="accordion_row_header"
                                  onClick={[Function]}
                                >
                                  <styles__AccordionColWrapper>
                                    <div
                                      className="styles__AccordionColWrapper-sc-19qwpvk-24 cVJzO"
                                    />
                                  </styles__AccordionColWrapper>
                                </div>
                              </styles__AccordionRow>
                              <styles__AccordionChildrenSection
                                accordionHeight=""
                              >
                                <div
                                  className="styles__AccordionChildrenSection-sc-19qwpvk-18 djglXG"
                                >
                                  <div>
                                    Hello Accordion
                                  </div>
                                </div>
                              </styles__AccordionChildrenSection>
                            </div>
                          </Col__StyledCol>
                        </Col>
                      </div>
                    </Row__StyledRow>
                  </Row>
                </div>
              </FlexGrid__StyledGrid>
            </FlexGrid>
          </div>
        </styles__Container>
      </AccordionSection>
    </div>
  </styles__AccordionMenu>
</Accordion>
`;
